debezium/src/main/scala/akka/stream/alpakka/debezium/scaladsl/DebeziumSource.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.debezium.scaladsl
+
+import java.util
+import java.util.concurrent.{ExecutorService, Executors}
+
+import akka.NotUsed
+import akka.actor.{ActorSystem, Props}
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.debezium.impl.CommitterActor
+import akka.stream.alpakka.debezium.{CommittableRecord, DebeziumSettings, SimpleRecord}
+import akka.stream.scaladsl._
+import akka.stream.{OverflowStrategy, QueueOfferResult}
+import io.debezium.embedded.Connect
+import io.debezium.engine.DebeziumEngine
+import org.apache.kafka.connect.source.SourceRecord
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Await, Future, Promise}
+
+object DebeziumSource {
+  def committable(settings: DebeziumSettings, executor: ExecutorService = Executors.newSingleThreadExecutor())(
+      implicit s: ActorSystem
+  ): Source[CommittableRecord, NotUsed] = {
+    val (queue, source) =
+      Source
+        .queue[List[CommittableRecord]](settings.maxInMemoryElements, OverflowStrategy.backpressure)
+        .mapConcat(identity)
+        .preMaterialize()
+
+    val configurations = settings.properties
+
+    val valueConverter = settings.internalValueConverter.getDeclaredConstructor().newInstance()
+    valueConverter.configure(configurations.asInstanceOf[java.util.Map[String, String]], false)
+
+    val engine = DebeziumEngine
+      .create[SourceRecord](classOf[Connect])
+      .using(configurations)
+      .notifying((records: util.List[SourceRecord], committer: DebeziumEngine.RecordCommitter[SourceRecord]) => {
+        val list = records.asScala.toList
+        val p = Promise[Unit]()
+        val committerActor = s.actorOf(Props(new CommitterActor(list, p, committer)))
+
+        def result: Future[Unit] =
+          queue
+            .offer(list.map(r => new CommittableRecord(r, committerActor, valueConverter)))
+            .flatMap {
+              case QueueOfferResult.Enqueued =>
+                p.future
+              case _ =>
+                Future.failed(new InterruptedException("error while enqueueing records"))
+            }(ExecutionContexts.sameThreadExecutionContext)
+
+        Await.result(result, settings.acknowledgmentTimeout)
```

Comment:
The only thing preventing Debezium to emit new events is this line of code. If we don't block here, it will continue to emit new elements, even if we don't commit the offset.

Reply:
Seems reasonable.  It's too bad how low-level we need to get to make this work.